tracer/build/_build/Build.Steps.cs

```diff
@@ -838,6 +865,51 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
                 workingDirectory / versionedName);
         });
 
+    Target ZipMonitoringHomeForAwsLambda => _ => _
+        .Unlisted()
+        .After(BuildTracerHomeForAwsLambda)
+        .Requires(() => IsLinux && (TargetPlatform == MSBuildTargetPlatform.x64 || TargetPlatform == ARM64TargetPlatform))
+        .Executes(() =>
+        {
+            var workingDirectory = ArtifactsDirectory / $"linux-{UnixArchitectureIdentifier}";
+            EnsureCleanDirectory(workingDirectory);
+
+            var arch = TargetPlatform == MSBuildTargetPlatform.x64 ? "amd64" : "arm64";
+            var archiveFile = workingDirectory / $"dd-trace-dotnet-aws-lambda-{arch}.zip";
+
+            // we want the tracer home directory to be in "/opt/datadog" in the AWS Lambda layer,
+            // so we need to nest it under a "datadog" folder in the zip file
+            var zipDirectory = MonitoringHomeDirectory / "lambda";
+            var datadogDirectory = zipDirectory / "datadog";
+            EnsureCleanDirectory(zipDirectory);
+            EnsureCleanDirectory(datadogDirectory);
+
+            foreach (var directory in Directory.GetDirectories(MonitoringHomeDirectory))
+            {
+                var directoryName = Path.GetFileName(directory);
+
+                if (directoryName != "lambda")
+                {
+                    Logger.Information("Moving {Source} to {Destination}", directory, datadogDirectory / directoryName);
+                    Directory.Move(directory, datadogDirectory / directoryName);
```

Comment:
Copy, don't move, so that this is idempotent 😄

Reply:
Lol, you wouldn't believe this but I spent an entire day trying different things here, including `Directory.Move`/`Copy` and the Nuke equivalents. Since I'm changing the output path anyway (due to another comment from you), I think I can add the `datadog` subdirectory there instead of copying stuff around.

Reply:
fixing this, thanks